Reader: Fix minor issues

wargcm opened the pull request:
## Description

Fixes:

- Background colors for the new design
- An out-of-bounds error when the table view is refreshing

## Testing

### Out-of-bounds crash

I'm not sure exactly how to reproduce the out-of-bounds error, it just happens randomly. For that, a manual code review should be enough. Since `i` should never be equal to the `count` of the array as that would always result in a crash.

### Background colors

- Launch Jetpack and login
- Enable the Reader Improvements v1 feature flag
- Navigate to the reader
- Tap on `Discover`
- 🔎 Scroll through and verify each card has the correct background color
- Switch to the opposite theming mode (light/dark)
- 🔎 Scroll through and verify each card has the correct background color

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

wargcm committed:
Merge branch 'trunk' into fix/reader-minor-issues

wargcm committed:
Fix reader out of bounds error on refresh

wargcm committed:
Update Reader cell background colors

dvdchr commented on WordPress/Classes/ViewRelated/Reader/ReaderRecommendedSiteCardCell.swift:
```diff
@@ -21,48 +24,30 @@ class ReaderRecommendedSiteCardCell: UITableViewCell {
-        let button = showLargeButton ? iPadFollowButton : followButton
-        let width = button?.frame.size.width ?? 0
-        infoTrailingConstraint.constant = width + Constants.buttonMargin
+        followButton.isHidden = !isLoggedIn
```
Just a note, there was a discussion with Chris in https://github.com/wordpress-mobile/WordPress-iOS/pull/21674#issuecomment-1747202728 where we're thinking of showing the Follow button for logged-out users as a way to get them to sign up. But perhaps for a later iteration.

dvdchr commented on WordPress/Classes/Utility/WPTableViewHandler.m:
```diff
@@ -151,7 +151,7 @@ - (void)refreshTableViewPreservingOffset
         newIndexPath = [self.resultsController indexPathForObject:obj];
 
-        if (i > visibleCellFrames.count) {
+        if (i >= visibleCellFrames.count) {
```
Good find 👍🏼

dvdchr reviewed (approved):
LGTM :shipit:

wargcm committed:
Merge branch 'feature/reader-recommended-tags-card' into fix/reader-minor-issues

wargcm committed:
Register both old and new post card cells

Updates the table view to register both the new and old post card
cells. This is done so that people switching the feature flag on/off
do not hit a crash.

wargcm committed:
Merge branch 'trunk' into fix/reader-minor-issues

wargcm commented:
@dvdchr Thanks for the review!

I'm going to enable auto-merge on this.
